src/main/java/jenkins/plugins/git/GitRepoSizeEstimator.java

```diff
@@ -0,0 +1,166 @@
+package jenkins.plugins.git;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.EnvVars;
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitTool;
+import hudson.plugins.git.util.GitUtils;
+import jenkins.model.Jenkins;
+import org.apache.commons.io.FileUtils;
+import org.jenkinsci.plugins.gitclient.Git;
+import org.jenkinsci.plugins.gitclient.GitClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * A class which allows {@link AbstractGitSCMSource} to estimate the size of a repository from a distance
+ * without requiring a local checkout.
+ */
+public class GitRepoSizeEstimator {
+
+    private long sizeOfRepo = 0L;
+    private String implementation;
+    private String gitTool;
+    public static final int SIZE_TO_SWITCH = 5000;
```

Comment:
Define the scale (Bytes, KB, MB, etc.) in comment, else it can be confusing 😅

Reply:
Good point, while writing the tests it has deemed to me that it might be cleaner to use a non-primitive data type instead of a primitive data type. 

The API might return varying sizes, we need to take that case into account as well.

Reply:
Yeah, you could perhaps have an abstract method which defaults to KB to use for any necessary conversions. Perhaps a different PR for something like that if we have time?

Reply:
@justinharringa Great suggestion, I have a seperate PR for this specification.